packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/private/helpers.ts

```diff
@@ -127,3 +138,36 @@ export function createProcessingConfig(
   }
   return undefined;
 }
+
+export function createBackupConfig(scope: Construct, role: iam.IRole, props?: S3BackupDestinationProps): DestinationBackupConfig | undefined {
+  if (!props) {
+    return undefined;
+  }
+  if (props.backupMode == null && !props.backupBucket) {
+    return undefined;
+  }
```

Comment:
We can't use `!props.backupMode` because when `props.backupMode` is set to `BackupMode.ALL` it will evaluate to `0` as that is the first entry in the enum. Do you want to do:

```suggestion
if (!props || (props.backupMode !== null && !props.backupBucket)) {
  return undefined;
}
```

I personally think the original way is more readable, even though it is more verbose.

Reply:
ugh javascript why. given that, I'd prefer
```suggestion
  if (!props || (props.backupMode === undefined && !props.backupBucket)) {
    return undefined;
  }
```